homeassistant/components/alphaess/sensor.py

```diff
@@ -0,0 +1,127 @@
+"""Alpha ESS Sensor definitions."""
+import datetime
+import logging
+
+from homeassistant.components.sensor import (
+    STATE_CLASS_MEASUREMENT,
+    STATE_CLASS_TOTAL_INCREASING,
+    SensorEntity,
+)
+from homeassistant.const import (
+    ATTR_IDENTIFIERS,
+    ATTR_MANUFACTURER,
+    ATTR_MODEL,
+    ATTR_NAME,
+    DEVICE_CLASS_BATTERY,
+    DEVICE_CLASS_ENERGY,
+    ENERGY_KILO_WATT_HOUR,
+    PERCENTAGE,
+)
+from homeassistant.helpers.device_registry import DeviceEntryType
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import ATTR_ENTRY_TYPE, DOMAIN
+
+_LOGGER: logging.Logger = logging.getLogger(__package__)
+
+
+async def async_setup_entry(hass, entry, async_add_entities) -> None:
+    """Defer sensor setup to the shared sensor module."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    for invertor in coordinator.data:
+        serial = invertor["sys_sn"]
+        async_add_entities(
+            [
+                AlphaESSSensor(coordinator, entry, serial, "Solar Production"),
```

Comment:
Did you look at creating `SensorEntityDescription`? This will help you define the entities in a nice way, where you don't have to add a lot of if statements in your actual sensor implementation.

See tuya/overkiz/shelly platforms for an example.

Reply:
I've just made the update to use SensorEntityDescription.  Thanks for the examples.